cli/src/plugins/javadoc/index.ts

```diff
@@ -164,8 +178,10 @@ function processParam(param: AnyType): string {
     param.simpleTypeName === "E" // TODO: Figure out how to filter for E implements RealmModel a bit... better
   ) {
     return "RealmModel";
-  } else if (param.simpleTypeName == "Callback") {
+  } else if (param.qualifiedTypeName == "io.realm.mongodb.App.Callback") {
     return "App.Callback";
+  } else if (param.qualifiedTypeName == "io.realm.Realm.Callback") {
```

Comment:
This shouldn't really have realm-specific stuff hardcoded in. Instead, find a way to have the user pass in a list of exceptions like these from the command line

Reply:
A fair point. I'm going to revisit this and try to just add a `enclosingclass.simpleName`.`class.simpleName` option to deal with some of these.

Reply:
I revisited this, and it turns out, by generating every unique combination of qualified name, simple name, name, and... some extra generic java magic, I'm able to do this without any realm-specific overrides!

It does involve some serious recursion... but I'll live.

Reply:
What if it were looked up on demand to avoid needing to generate a massive list? For example make another attempt to resolve links at the final stage by guessing if #someMethod actually means someClass.someMethod etc